virtdata-lib-basics/src/test/java/io/nosqlbench/virtdata/library/basics/tests/libraryimpl/BasicDataMappersTest.java

```diff
@@ -52,14 +51,17 @@ public void testToDateInstantiator() throws Exception {
         assertThat(dataMapper.get()).isNotNull();
         Date d1 = dataMapper.get().get(1);
         Date d2 = dataMapper.get().get(2);
+        assertThat(d2).isAfter(d1);
     }
 
     @Test
     public void testToDateBucketInstantiator() throws Exception {
         Optional<DataMapper<Date>> dataMapper = VirtData.getOptionalMapper("ToDate(1000,10000)");
         assertThat(dataMapper).isNotNull();
         assertThat(dataMapper.get()).isNotNull();
-        assertThat(dataMapper.get().get(1).after(new Date(1)));
```

Comment:
this returned false but no assertion was ever executed... is the new code below the expected logic?

Reply:
The semantics of this function are a bit muddled. I'm adding unit tests and appropriate documentation. The second param is indicated as a "repeat" factor, but It looks sus. I'll clean it up with unit tests.